### PR TITLE
Change docs linting from `markdownlint-cli` to `markdownlint-cli2`

### LINKED ISSUE #1250: Switch from `markdownlint-cli` to `markdownlint-cli2`
### What should be done?

We should switch from https://github.com/igorshubovych/markdownlint-cli to https://github.com/DavidAnson/markdownlint-cli2 because it should be faster.

### Where?

* https://github.com/FerretDB/FerretDB/blob/main/build/deps/markdownlint.Dockerfile
* https://github.com/FerretDB/FerretDB/blob/main/Taskfile.yml
* https://github.com/FerretDB/FerretDB/blob/main/.textlintignore
* https://github.com/FerretDB/FerretDB/blob/main/.markdownlint.yml

### PR DESCRIPTION
**Pull request by codingmickey**
# Description

Closes #1250.

## Readiness checklist

* [ ] I added tests for new functionality or bugfixes.
* [ ] I ran `task all`, and it passed.
* [ ] I added/updated comments for both exported and unexported top-level declarations (functions, types, etc).
* [ ] I checked comments rendering with `task godocs`.
* [ ] I ensured that the title is good enough for the changelog.
* [ ] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Assignee, Labels, Project and project's Sprint fields.
* [ ] I marked all done items in this checklist.

### CODE REVIEW

File: build/deps/markdownlint.Dockerfile [hunk 1]
```diff
@@ -1 +1 @@
-FROM ghcr.io/igorshubovych/markdownlint-cli:v0.32.2
+FROM davidanson/markdownlint-cli2:latest
```
Comment (AlekSi) [inline build/deps/markdownlint.Dockerfile:1]: Please add a new line

File: .markdownlint.yml [hunk 2]
```diff
@@ -1,13 +1,16 @@
...
-MD004: {style: "asterisk"}
+MD003:
+  style: "atx"
+MD004:
+  style: "asterisk"
 MD006: false # deprecated
 MD013: false # we use https://sembr.org that could lead to long lines
-MD029: {style: ordered}
+MD029:
+  style: "ordered"
```
Comment (AlekSi) [inline .markdownlint.yml:16]: Please keep formatting the same

File: Taskfile.yml [hunk 3]
```diff
@@ -17,7 +17,7 @@ tasks:
     deps: [all]
 
   all:
-    desc: "Generate, format, build, test and lint code and documentation"
+    desc: 'Generate, format, build, test and lint code and documentation'
```
Comment (AlekSi) [inline Taskfile.yml:20]: Please keep formatting the same

File: .markdownlint.yml (hunk 2 above)
Comment (codingmickey) [reply .markdownlint.yml:16]: this is the same formatting just the syntax is changed according to the newer version

Comment (AlekSi) [reply .markdownlint.yml:16]: Both old and new lines represent the same YAML object; the former just uses JSON-like inline formatting. Let's keep it that way to minimize unrelated changes in this PR.

File: Taskfile.yml [hunk 4]
```diff
@@ -98,7 +98,7 @@ tasks:
       - docker-compose logs --follow
 
   env-pull:
-    desc: "Pull development environment's Docker images"
+    desc: "Pull development environment"s Docker images"
```
Comment (AlekSi) [inline Taskfile.yml:101]: Please check the diff of that PR and revert all unrelated formatting changes. This line, in particular, has invalid syntax because there are three quotation marks.

**Review by AlekSi — CHANGES_REQUESTED** [review]
